src/CoverageTools.jl

```diff
@@ -186,8 +186,12 @@ module CoverageTools
             lineoffset = searchsortedlast(linepos, pos - 1) - 1
 
             # now we can parse the next chunk of the input
-            ast, pos = Meta.parse(content, pos)
+            ast, pos = Meta.parse(content, pos; raise=false)
             isa(ast, Expr) || continue
+            if ast.head ∈ (:error, :incomplete)
+                lineoffset = searchsortedlast(linepos, pos - 1) - 1
+                throw(Base.Meta.ParseError("parsing error in $(fc.filename), successfully handled up to line $lineoffset"))
```

Comment:
I would avoid using the identifier `lineoffset` here as that's used before to store a 0-based offset while we want to print a 1-based line number to the user).

Also, I think we could do even better by extracting the error message.
```
julia> str = """for x =

        end"""; Meta.parse(str, 1, raise = false)
(:($(Expr(:error, "unexpected \"end\""))), 13)


julia> str = """for x =

        """; Meta.parse(str, 1, raise = false)
(:($(Expr(:incomplete, "incomplete: premature end of input"))), 10)
```

So how about this:
```suggestion
                line = searchsortedlast(linepos, pos - 1)
                throw(Base.Meta.ParseError("parsing error in $(fc.filename):$line: $(ast.args[1])"))
```
Or if we want to match the message Julia prints more closely:
```suggestion
                line = searchsortedlast(linepos, pos - 1)
                throw(Base.Meta.ParseError("syntax: $(ast.args[1]) at $(fc.filename):$line"))
```

Well, except the error message actually can contain a "filename" already... ugh:
```
julia> str = """
       function parseerr()
           s = 0
           for i = [1,2,3]   # this line has a parsing error
               s += i
           end
           return s
       """;

julia> Meta.parse(str, 1, raise = false)
(:($(Expr(:incomplete, "incomplete: \"function\" at none:1 requires end"))), 121)
```
Interestingly, in the Julia master branch the lower-level parsing functions all have kwargs `filename` and `index` -- but `parse` doesn't expose them. Guess it's time for yet another PR :-).